sentry-kotlin-compiler-plugin/src/test/kotlin/io/sentry/compose/JetpackComposeInstrumentationTest.kt

```diff
@@ -0,0 +1,291 @@
+
+        fun execute(
+            compilation: KotlinCompilation.Result,
+            className: String = "io.sentry.samples.Example",
+            method: String,
+            methodArgTypes: List<String> = emptyList(),
+            methodArgs: List<Any> = emptyList()
+        ): List<String> {
+            // inject a callback into our fake modifier
+            val tags = mutableListOf<String>()
+            try {
+                val fakeModifierClass =
+                    compilation.classLoader.loadClass("io.sentry.compose.SentryModifierKt")
+                val setCallbackMethod =
+                    fakeModifierClass.getMethod("setCallback", Function1::class.java)
+                setCallbackMethod.invoke(fakeModifierClass, { tag: String ->
+                    tags.add(tag)
+                })
+            } catch (ex: ClassNotFoundException) {
+                // no-op
+            }
+
+            val kClazz = compilation.classLoader.loadClass(className)
+            val exampleObj = kClazz.getDeclaredConstructor().newInstance()
+
+            val argClasses =
+                methodArgTypes.map { compilation.classLoader.loadClass(it) }.toTypedArray()
+            val args = methodArgs.toTypedArray()
+            kClazz.getMethod(method, *argClasses).invoke(exampleObj, *args)
+
+            return tags
+        }
+    }
+
+    @Test
+    fun `When no modifier is present, inject sentry modifier`() {
+        val kotlinSource = SourceFile.kotlin(
+            name = "Example.kt",
+            contents = """
+            package io.sentry.samples
+            import androidx.compose.runtime.Composable
+            import io.sentry.compose.ComposableFunction
+
+            class Example {
+                @Composable
+                fun NoModifier() {
+                    ComposableFunction(
+                        text = "No Modifier Argument"
+                    )
+                }
+            }
+            """.trimIndent()
+        )
+
+        val fixture = Fixture()
+
+        val compilation = fixture.compileFile(kotlinSource)
+        assert(compilation.exitCode == KotlinCompilation.ExitCode.OK)
+
+        val tags = fixture.execute(compilation, method = "NoModifier")
+        assertEquals(1, tags.size)
+        assertEquals("NoModifier", tags[0])
+    }
+
+    @Test
+    fun `Modifier Companion calls are replaced with sentry modifier`() {
+        val kotlinSource = SourceFile.kotlin(
+            name = "Example.kt",
+            contents = """
+            package io.sentry.samples
+
+            import androidx.compose.foundation.layout.fillMaxSize
+            import androidx.compose.foundation.layout.padding
+            import androidx.compose.ui.unit.dp
+            import androidx.compose.runtime.Composable
+            import androidx.compose.ui.Modifier
+            import io.sentry.compose.ComposableFunction
+
+            class Example {
+                @Composable
+                fun ExistingModifier() {
```

Comment:
`l`: I guess it'd be nice to add the expected modified code snippet as a comment here so it's easier to understand what are we testing. This also applies for all tests in this class